### ISSUE TITLE
simplify readfile() et al, extract into functions, remove MACOS_CONVERT

### ISSUE THREAD

**Issue by aktau** [neovim/neovim#1374]
`readfile()` is a very important but mastodont-like function (1762 lines) at the moment of speaking. I've been trying to make more sense of it, but its structure is a bit hard to follow. Among other things because it likes to use internal goto's. I have no problem with goto's but in a 1762-line function it's a bit much.

So I had the idea of chopping it down to make it more understandable. One small win was that (I think) we can remove the parts that reference `try_mac` and `MACOS_CONVERT` since we don't support MacOS 9. That said, some people might want to edit files from that era, so I'm not sure if this would be broadly supported.

But extracting some functionality into functions would help readability and not impair functionality.

Thoughts?

**Comment by justinmk** [comment]
Nothing here that needs tracking.
